### ISSUE TITLE
Crash during reset when getting MD members

### ISSUE THREAD

**Issue by vojtechtrefny** [storaged-project/blivet#1270]
This code change seems to be reopened about DiskDevice get members crashed bug
![image](https://github.com/user-attachments/assets/372c1ac4-d71b-45e0-bd61-8116d9d0dba2)

I don't understand why the path check for md is removed here:
![image](https://github.com/user-attachments/assets/e8b748fd-0bfe-4bb4-851e-072c96e6c2ca)

This is the data I observed:
![image](https://github.com/user-attachments/assets/4d41dc84-51a0-46d6-9228-70d01f57bd93)

The program crashed in storage.reset()

_Originally posted by @Yurri-huang in https://github.com/storaged-project/blivet/issues/888#issuecomment-2298594608_

**Comment by vojtechtrefny** [comment]
> This code change seems to be reopened about DiskDevice get members crashed bug

The issue seems to be that we expect all MD BIOS RAIDS to have two levels of RAID devices -- one regular MD device and one container so here we try to get the member devices (the disks) by looking one level "below" the MD device. Looking at the `lsblk` output here, there is just one MD device.

@Yurri-huang can you please share `mdadm --detail /dev/md127` and `mdadm --detail /dev/md126`? Also if you are running the code with logging enabled, can you please share the full log as well. Thanks.

> I don't understand why the path check for md is removed here:

It's moved to the first `if` -- the added `device_is_md` check does the same check which was removed.

**Comment by Yurri-huang** [comment]
> > This code change seems to be reopened about DiskDevice get members crashed bug
> 
> The issue seems to be that we expect all MD BIOS RAIDS to have two levels of RAID devices -- one regular MD device and one container so here we try to get the member devices (the disks) by looking one level "below" the MD device. Looking at the output here, there is just one MD device.`lsblk`
> 
> @Yurri-huang can you please share and ? Also if you are running the code with logging enabled, can you please share the full log as well. Thanks.`mdadm --detail /dev/md127``mdadm --detail /dev/md126`
> 
> > I don't understand why the path check for md is removed here:
> 
> It's moved to the first -- the added check does the same check which was removed.`if``device_is_md`

lsblk: 
![image](https://github.com/user-attachments/assets/d8622d4d-f610-4e4f-8601-8a0e1ea91d81)

mdadm:
![image](https://github.com/user-attachments/assets/6eb9b05a-2480-417f-962b-65c9ea34fff7)

**Comment by Yurri-huang** [comment]
I don't have a log on blivet, I don't know how to open a log (I'm a newbie), if you need a log, I can turn it on

**Comment by vojtechtrefny** [comment]
You can enable by calling `blivet.util.set_up_logging()`:

```
import blivet
blivet.util.set_up_logging()
b=blivet.Blivet()
b.reset()
```

The log file will be saved to `/tmp/blivet.log`

**Comment by Yurri-huang** [comment]
[blivet-testdata.log](https://github.com/user-attachments/files/16675676/blivet-testdata.log)
[blivet.log](https://github.com/user-attachments/files/16675677/blivet.log)

**Comment by Yurri-huang** [comment]
'ID_FS_TYPE=isw_raid_member' was not correctly recognized.

I performed the return name operation for 'isw_raid_member' in the same way as' linux_raid_comember ', and it won't crash.

```patch

        elif device_get_format(udev_info) == "linux_raid_member":
            # MD RAID member -> use SYS_NAME
            name = udev_info["SYS_NAME"]
+        elif device_get_format(udev_info) == "isw_raid_member":
+           name = udev_info["SYS_NAME"]
        elif device_is_partition(udev_info):
            # partition on RAID -> construct name from MD_DEVNAME + partition number
            # for partitions on named RAID we want to use the raid name, not
            # the node, e.g. "raid1" instead of "md127p1"
            partnum = udev_info["ID_PART_ENTRY_NUMBER"]
```

Can I submit this patch?

**Comment by vojtechtrefny** [comment]
> Can I submit this patch?

Yes please, thank you.

**Comment by vojtechtrefny** [comment]
Closed via #1271
